scripts/initializers/order.js

```diff
@@ -68,14 +73,15 @@ async function handleUserOrdersRedirects(
   returnRef,
   isTokenProvided,
   langDefinitions,
+  orderNumber,
 ) {
   let targetPath = null;
 
   events.on('order/error', () => {
     if (checkIsAuthenticated()) {
       window.location.href = CUSTOMER_ORDERS_PATH;
     } else if (isTokenProvided) {
-      window.location.href = ORDER_STATUS_PATH;
+      window.location.href = orderNumber ? ORDER_STATUS_PATH : `${ORDER_STATUS_PATH}?orderRef=${orderNumber}`;
```

Comment:
The conditional seems backwards. If orderNumber exists, don't use it?

Reply:
Thanks, you are right 🙈